### PR TITLE
Allow atomic operations on unshared memories

### PR DESCRIPTION
**Pull request by tlively**
As discussed in the CG on November 12, 2019 and in #144. Specifically,
all atomic accesses are now allowed to validate and execute normally
on unshared memories and wait operations trap when used with unshared
memories.

The PR updates Overview.md and makes a best-effort
attempt at updating the spec, making no changes when there is
already a TODO for updating spec text. It also adds new TODO comments
to the reference interpreter where changes will have to be made.

### CODE REVIEW

File: document/core/valid/instructions.rst [hunk 1]
```diff
@@ -440,8 +440,6 @@ Atomic Memory Instructions
 
 * Let :math:`\limits~\share` be the :ref:`memory type <syntax-memtype>` :math:`C.\CMEMS[0]`.
 
-* The sharedness :math:`\share` must be |MSHARED|.
```
Comment (rossberg) [inline document/core/valid/instructions.rst]: You'll also need to adjust the respective premise in the formal rule below to keep it in sync, e.g., by changing `C.\CMEMS[0] = \limits~\MSHARED` to `C.\CMEMS[0] = \memtype` (here and elsewhere).

**Comment by tlively** [comment]
Sounds good, I'll add a vote to the agenda for the next meeting 👍

**Comment by tlively** [comment]
@rossberg I just want to confirm that the latest changes are what you had in mind.

**Comment by tlively** [comment]
I'll merge this now, but of course I am happy to fix any mistakes or holes in follow-up PRs.

**Pull request merged**